src/main/java/Storage.java

```diff
@@ -0,0 +1,151 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.ToDo;
+
+/**
+ * Storage deals with loading tasks from the file and saving tasks in the file.
+ *
+ * @author Sebastian Tay
+ */
+public class Storage {
+    private final String dir = "/VedaMissions";
+    private final String storagePath = "/VedaMissions/Missions.txt";
+    private final File storageFile = new File(storagePath);
+
+    /**
+     * Adds file into the directory specified.
+     *
+     * @return whether file has been successfully added into the hard disk.
+     */
+    public boolean addFile() {
+        File directory = new File(dir);
+
+        try {
+            if (directory.mkdir()) {
+                storageFile.createNewFile();
+
+                return true;
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred while trying to create the file.");
+        }
+
+        return false;
+    }
+
+    /**
+     * Indicates if the saved file is existing.
+     *
+     * @return true if the saved file does exist, false otherwise.
+     */
+    public boolean checkFileExists() {
+        return storageFile.exists();
+    }
+
+    /**
+     * Returns the stored task list from the hard disk.
+     *
+     * @return an ArrayList containing the tasks stored in the hard disk.
+     * @throws FileNotFoundException If the storageFile is not located in its rightful directory.
+     */
+    public ArrayList<Task> retrieveTasks() throws FileNotFoundException {
+
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        Scanner sc = new Scanner(storageFile);
+
+        while (sc.hasNextLine()) {
+            final String task = sc.nextLine(); //Each task is listed in a single line
+
+            try {
+                char type = task.charAt(0);
+                String[] descriptions = task.split("::"); //:: demarcates a different field
+
+                switch (type) {
+                case 'T':
+                    tasks.add(new ToDo(descriptions[2], descriptions[1].matches("1")));
+                    break;
+
+                case 'D':
+                    //Events.Deadline
+                    tasks.add(new Deadline(descriptions[2], descriptions[3], descriptions[1].matches("1")));
+                    break;
+
+                case 'E':
+                    //Events.Event
+                    tasks.add(new Event(descriptions[2], descriptions[3], descriptions[1].matches("1")));
+                    break;
+
+                default:
+                    System.out.println("Wrong file format");
+                }
```

Comment:
Good job for correct formatting for case clauses!

Reply:
Thanks!